mslice/scripting/helperfunctions.py

```diff
@@ -0,0 +1,138 @@
+            add_cut_plot_statements(script_lines, plot_handler)
+
+        script_lines.append('mc.Show()\n')
+
+    return script_lines
+
+
+def add_slice_plot_statements(script_lines, plot_handler):
+    script_lines.append('slice_ws = mc.Slice(ws)\n\n')
+    script_lines.append('fig = plt.gcf()\n')
+    script_lines.append('ax = fig.add_subplot(111, projection=\'mslice\')\n')
+
+    if plot_handler.default_options['intensity'] is True:
+        intensity = _function_to_intensity[plot_handler.default_options['intensity_method']]
+        script_lines.append(
+            'mesh = ax.pcolormesh(slice_ws, cmap=\'{}\', intensity=\'{}\', temperature={})\n'.format(
+                plot_handler._slice_plotter_presenter._slice_cache[plot_handler.ws_name].colourmap, intensity,
+                plot_handler.default_options['temp'] if plot_handler.default_options['temp_dependent'] else None))
+    else:
+        script_lines.append(
+            'mesh = ax.pcolormesh(slice_ws, cmap=\'{}\')\n'.format(
+                plot_handler._slice_plotter_presenter._slice_cache[plot_handler.ws_name].colourmap))
+
+    script_lines.append('mesh.set_clim({}, {})\n'.format(*plot_handler.colorbar_range))
+    script_lines.append(
+        'mesh.set_norm(colors.LogNorm({}, {}))\n'.format(
+            0.001 + plot_handler.colorbar_range[0] if plot_handler.colorbar_range[0] == 0 else
+            plot_handler.colorbar_range[0], plot_handler.colorbar_range[1]) if plot_handler.colorbar_log is True else ''
+    )
+
+    script_lines.append('cb = plt.colorbar(mesh, ax=ax)\n')
+    script_lines.append('cb.set_label(\'{}\', labelpad=20, rotation=270, picker=5)\n'.format(
+        plot_handler.colorbar_label))
+
+    script_lines.append('ax.set_title(\'{}\')\n'.format(plot_handler.title)
+                        if plot_handler.is_changed('title') else '')
+
+    script_lines.append('ax.set_ylabel(\'{}\')\n'.format(plot_handler.y_label)
+                        if plot_handler.is_changed('y_label') else '')
+    script_lines.append('ax.set_xlabel(\'{}\')\n'.format(plot_handler.x_label)
+                        if plot_handler.is_changed('x_label') else '')
+
+    script_lines.append('ax.grid({}, axis=\'y\')\n'.format(plot_handler.y_grid)
+                        if plot_handler.is_changed('y_grid') else '')
+    script_lines.append('ax.grid({}, axis=\'x\')\n'.format(plot_handler.x_grid)
+                        if plot_handler.is_changed('x_grid') else '')
+
+    script_lines.append('ax.set_ylim(bottom={}, top={})\n'.format(*plot_handler.y_range)
+                        if plot_handler.is_changed('y_range') else '')
+    script_lines.append('ax.set_xlim(left={}, right={})\n'.format(*plot_handler.x_range)
+                        if plot_handler.is_changed('x_range') else '')
+
+
+def add_overplot_statements(script_lines, plot_handler):
+    ax = plot_handler._canvas.figure.gca()
+    line_artists = [artist for artist in ax.get_children() if isinstance(artist, Line2D)]
```

Comment:
The lines are all stored in the `ax.lines` attribute so we can avoid having to do this loop here.